### ISSUE TITLE
ENH: Add a site-wide glossary and index of acronyms

### ISSUE THREAD

**Issue by rossbar** [scientific-python/scientific-python.org#77]
I'm thinking something along the lines of what sphinx has with the `:term:` and `:abbr:` roles, where the former links the term to a glossary entry, and the latter adds a pop-up box that shows the full name associated with an acronym when a user hovers over the link with their cursor. 

I haven't looked closely, so it's possible the `mainroad` theme already supports some or all of this. If not, maybe this functionality could be added via shortcodes.

Not a critical feature by any means, I just thought it'd be a nice improvement given the number of new terms and acronyms that are on the site.

**Comment by stefanv** [comment]
Would this automatically be added to each page, or would you manually have to add it for each keyword?  The prior would make sense to me, but may be a tad trickier to implement.  Probably not too hard to do if we pre-process the text, or add a manual filter to Hugo in our template.

**Comment by rossbar** [comment]
>  or would you manually have to add it for each keyword? The prior would make sense to me, but may be a tad trickier to implement.

Good point - I was locked into the role-markup mindset, but automatically cross-linking glossary terms and abbreviations would certainly be nice. :+1: to this idea!
